lib/src/mui_suggestors/components/mui_inline_alert_migrator.dart

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+
+import 'package:over_react_codemod/src/mui_suggestors/constants.dart';
+import 'package:over_react_codemod/src/util/component_usage.dart';
+import 'package:over_react_codemod/src/util/component_usage_migrator.dart';
+
+import 'mui_migrator.dart';
+
+class MuiInlineAlertMigrator extends ComponentUsageMigrator
+    with MuiMigrator, AlertDisplayPropsMigrator {
+  static const toastAlertProps = [
+    'dismissAfter',
+    'initiallyShown',
+    'isDismissible',
+    'onDidHide',
+    'onDidShow',
+    'onWillHide',
+    'onWillShow',
+    'transition',
+    'transitionCount',
+    'transitionInCount',
+    'transitionOutCount',
+  ];
+
+  @override
+  bool shouldMigrateUsage(FluentComponentUsage usage) =>
+      usesWsdFactory(usage, 'Alert') &&
+      // If the alert has props related to the `toast` variant, don't try to
+      // migrate it.
+      !usage.cascadedProps
+          .any((prop) => toastAlertProps.contains(prop.name.name));
+
+  @override
+  void migrateUsage(FluentComponentUsage usage) {
+    super.migrateUsage(usage);
+
+    yieldPatchOverNode('$muiNs.Alert', usage.factory!);
+
+    handleCascadedPropsByName(usage, {
+      'heading': (p) => migrateAlertHeading(usage, p),
+      'headingProps': yieldPropManualMigratePatch,
+      'skin': migrateAlertSkin,
+      'size': migrateAlertSize,
+    });
+  }
+}
+
+mixin AlertDisplayPropsMigrator on ComponentUsageMigrator {
+  void migrateAlertSkin(PropAssignment prop) {
+    final rhs = prop.rightHandSide;
+
+    final severityFromWsdSkin = mapWsdConstant(rhs, const {
+      'AlertSkin.DEFAULT': '$muiNs.AlertSeverity.info',
+      'AlertSkin.SUCCESS': '$muiNs.AlertSeverity.success',
+      'AlertSkin.WARNING': '$muiNs.AlertSeverity.warning',
+      'AlertSkin.DANGER': '$muiNs.AlertSeverity.error',
+    });
+
+    if (severityFromWsdSkin != null) {
+      yieldPropPatch(
+        prop,
+        newName: 'severity',
+        newRhs: severityFromWsdSkin,
+      );
+      return;
+    }
+
+    final usesInverseSkin =
+        isWsdStaticConstant(prop.rightHandSide, 'AlertSkin.INVERSE');
+    var migrateVariantToGray = usesInverseSkin ||
+        isWsdStaticConstant(prop.rightHandSide, 'AlertSkin.GRAY');
+
+    mapWsdConstant(rhs, const {
+      'AlertSkin.GRAY': '$muiNs.AlertVariant.wsdGray',
+    });
+
+    if (migrateVariantToGray) {
+      if (usesInverseSkin) {
+        yieldPropFixmePatch(prop,
+            'this prop was converted from the INVERSE skin and should be double checked');
+      }
+
+      yieldPropPatch(
+        prop,
+        newName: 'variant',
+        newRhs: '$muiNs.AlertVariant.wsdGray',
+      );
+      return;
+    }
+
+    yieldPropManualMigratePatch(prop);
+  }
+
+  void migrateAlertHeading(FluentComponentUsage usage, PropAssignment prop) {
+    final propValue = prop.rightHandSide;
+
+    yieldRemovePropPatch(prop);
+    yieldAddChildPatch(usage, '$muiNs.AlertTitle()($propValue)');
```

Comment:
`$propValue` uses the toString value of the AstNode; can we use the original source instead?
```suggestion
    yieldAddChildPatch(usage, '$muiNs.AlertTitle()(${context.sourceFor(propValue)})');
```

([More context on why to avoid toString](https://github.com/Workiva/over_react_codemod/blob/master/README.md#:~:text=Avoid%20using%20AstNode.toSource()%20(or%20.toString())%20outside%20of%20tests))